rust/abacus-base/src/settings/chains.rs

```diff
@@ -111,6 +111,34 @@ impl<T> ChainSetup<T> {
             .expect("could not parse finality_blocks")
     }
 
+    /// Try to convert the chain settings into an AbacusProvider.
+    pub async fn try_into_provider(
+        &self,
+        metrics: &CoreMetrics,
+    ) -> eyre::Result<Box<dyn AbacusProvider>> {
+        let builder = AbacusProviderBuilder {};
+        let metrics_conf = {
+            let mut cfg = self.metrics_conf.clone();
+
+            if cfg.chain.is_none() {
+                cfg.chain = Some(ChainInfo {
+                    name: Some(self.name.clone()),
+                });
+            }
+
+            cfg
+        };
+
+        self.build(
+            "0x0000000000000000000000000000000000000000",
```

Comment:
The entire building of these is a mess that we are going to need to address anyway. If you think an option makes it nicer I will do that. I was avoiding it since it felt like a lot of changes that were not really needed but it would be more correct.

Reply:
I sort of take this back, it really does not feel right this way and adds a lot of boilerplate. Ends up having to propagate all the way to the chain locator. I can easily enough revert the commit if you prefer.

Reply:
Chatted offline, went with the suggested approach of
```
let address: String = match &self.chain {
  ChainConf::Ethereum(_) => ethers::types::Address::zero().encode_hex()
};
```